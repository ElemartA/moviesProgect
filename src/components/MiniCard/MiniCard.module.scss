.wrap {
  width: 350px;
  height: auto;
  color: black;
  font-weight: 600;
  font-size: 18px;
  margin-left: 1rem;
  box-sizing: border-box;
  opacity: 0.7;
 
  .wrap_card {
    width: 27.5rem;
    margin-bottom: 8px;
    background-color: rgba(204, 175, 220, 0.931);
    .wrap_img {
      box-sizing: border-box;
      width: 100%;
      border: solid #b66be4 5px;
      padding: 5px;
      background: cover;
    }
  }

  .file {
    width: 90%;
    margin: 0 auto;
    height: auto;
  }

  .wrap_block {
    width: 10rem;
    height: auto;
    .wrap_img {
      box-sizing: border-box;
      width: 100%;
      border: solid #5b6dcd 5px;
      padding: 5px;
      background: cover;
    }
    .wrap_hidden {
      width: 0;
    }
  }
}
