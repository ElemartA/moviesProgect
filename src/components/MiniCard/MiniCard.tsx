import React, { FunctionComponent } from "react";
import { TMiniCard } from "../../types/miniCardTypes";
import s from "./MiniCard.module.scss";

type MyProps = {
  value2: TMiniCard[];
};

const MiniCard: FunctionComponent<MyProps> = ({ value2 }) => (
  <div data-testid="miniCard" className={s.wrap}>
    <div>
      {value2.map((el, i) => (
        <div key={i} className={s.wrap_card}>
          {Object.values(el).map((elem, index) =>
            `${elem}`.includes("blob") ? (
              <img
                key={index}
                alt={"avatar"}
                className={s.wrap_img}
                src={`${elem}`}
              ></img>
            ) : (
              <div key={index} className={s.wrap_item}>
                {`${elem}`}
              </div>
            )
          )}
        </div>
      ))}
    </div>
  </div>
);

export default MiniCard;
