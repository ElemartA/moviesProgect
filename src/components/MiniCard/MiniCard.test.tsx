import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import MiniCard from "./MiniCard";

describe("MiniCard", () => {
  it("renders MiniCard component", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <MiniCard value2={[]} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    screen.debug();
    expect(screen.getByTestId("miniCard")).toBeInTheDocument();
  });
});
