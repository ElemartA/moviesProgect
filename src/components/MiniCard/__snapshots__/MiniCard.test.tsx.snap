// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniCard renders MiniCard component 1`] = `
<DocumentFragment>
  <div
    class="wrap"
    data-testid="miniCard"
  >
    <div />
  </div>
</DocumentFragment>
`;
